### ISSUE TITLE
Get-AzureRmResource help missing existing params and includes non-existent params

### ISSUE THREAD

**Issue by jonsabo** [Azure/azure-powershell#3173]
### Cmdlet(s)
Get-AzureRmResource

### PowerShell Version
5.1.14393.223

### Module Version
3.1.0

### OS Version
10.0.14393.223

### Description
The help for `Get-AzureRmResource` includes some parameters that do not exist in the current version, and is missing some parameters that do exist. I ran `update-help` just to be very sure as well.

For example the help mentions these parameters that are not available:
```
Tag
Profile
Name
ParentResource
```

and does not mention:
```
ExtensionResourceName
ExtensionResourceType
ResourceId
```

Here's the parameters described in the help
```
> $(get-help Get-AzureRmResource -full).Parameters.Parameter | select Name

name             
----             
ResourceGroupName
ResourceType     
Tag              
Profile          
Name             
ParentResource   
ApiVersion          
```

And what is actually available
```
> get-command Get-AzureRmResource | select -ExpandProperty Parameters | select -ExpandProperty keys
ResourceId
ResourceName
ResourceType
ExtensionResourceName
ExtensionResourceType
ExpandProperties
IsCollection
Top
ODataQuery
ResourceGroupName
TenantLevel
ApiVersion
Pre
Verbose
Debug
ErrorAction
WarningAction
InformationAction
ErrorVariable
WarningVariable
InformationVariable
OutVariable
OutBuffer
PipelineVariable
```

**Comment by markcowl** [comment]
@ravbhatnagar @vivsriaus   Can you guys take a look?  This would be a great time to use the new platyPS help tools to make help editing easier.https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md

**Comment by MattAllison** [comment]
The lack of accurate help is a big problem.

Guys, this is killing me right now - I'm trying to train new staff on how to do things in ARM and one of the most basic commands doesn't work as advertised.

**Comment by ravbhatnagar** [comment]
@MattAllison @jonsabo - Apologies for this issue here and that its taken us long to fix. Please feel free to send a PR for this fix. Else we will address it in the upcoming release.

**Comment by cormacpayne** [comment]
This will be fixed in PR https://github.com/Azure/azure-powershell/pull/6008

**Comment by cormacpayne** [comment]
The fix for this should be available on May 7th in version 6.0.0 of `AzureRM` (version 6.0.0 of `AzureRM.Resources`). Please re-open this issue if the problem persists.
